src/v/kafka/server/request_context.h

```diff
@@ -275,6 +267,33 @@ class request_context {
         usage_mgr().add_bytes_sent(response_size - internal_bytes_sent);
     }
 
+    template<typename ResponseType>
+    ss::future<> log_response(ResponseType res) {
+        // Create the log message early because members _conn and _header are
+        // destroyed at the following co_await
+        auto msg = fmt::format(
+          "[{}:{}] sending {}:{} for {}, response {}",
+          _conn->client_host(),
+          _conn->client_port(),
+          ResponseType::api_type::key,
+          ResponseType::api_type::name,
+          _header.client_id,
+          res);
```

Comment:
In the case of responses like fetch this can still write massive/pii responses into the logs. Is there a way to prevent this? maybe `errored` should only return the error response codes and messages?

Reply:
It seems that the concern here is about the content that is logged but there are likely other places besides`request_context` where this concern is valid. Additionally, the request for this PR is about logging errored responses, not changing the content of the logs. Perhaps this comment-thread should be a separate project or ticket?